src/pages/Send/components/RecipientPicker/RecipientPicker.js

```diff
@@ -0,0 +1,432 @@
+// @flow strict
+
+import React, {
+  Component,
+} from 'react'
+import { t } from 'ttag'
+
+import escapeRegExp from 'utils/regexp/escapeRegExp'
+
+import {
+  JPickerBody,
+  JPickerList,
+  JPickerCurrent,
+  NotFoundItem,
+} from 'components/base/JPicker'
+import { JIcon } from 'components/base'
+import {
+  startsWithOrEndsWith,
+  checkAddressPartValid,
+} from 'utils/address'
+
+import { Empty } from './Tabs/Empty'
+import { ContactItem } from './ContactItem/ContactItem'
+import { ContactIcon } from './ContactIcon/ContactIcon'
+import { WalletList } from './WalletList/WalletList'
+import { QuickSendItem } from './QuickSendItem/QuickSendItem'
+import {
+  Tabs,
+  type Tab,
+} from './Tabs/Tabs'
+
+type Contact = {|
+  +name?: string,
+  +description?: string,
+  +address: Address,
+|}
+
+export type RecipientPickerWalletAddress = {|
+  address: Address,
+  name: string,
+  fiatBalance?: string,
+|}
+
+export type RecipientPickerWallet = {|
+  id: WalletId,
+  type: 'address' | 'mnemonic' | 'read-only',
+  name: string,
+  addresses: RecipientPickerWalletAddress[],
+|}
+
+function filterContacts(
+  contacts: Contact[],
+  searchQuery: string,
+): Contact[] {
+  const query: string = searchQuery.trim()
+  const searchRe: RegExp = new RegExp(escapeRegExp(query), 'ig')
+
+  return !query ? contacts : contacts.reduce((
+    result,
+    contact,
+  ) => {
+    const {
+      name,
+      description,
+      address,
+    } = contact
+
+    const isFound: boolean =
+      (name && searchRe.test(name)) ||
+      (description && searchRe.test(description)) ||
+      startsWithOrEndsWith(address, query)
+
+    return !isFound ? result : [
+      ...result,
+      contact,
+    ]
+  }, [])
+}
+
+function getAddressName(address: RecipientPickerWalletAddress, addressIndex: number) {
```

Comment:
Maybe move it to `utils`? It looks like it would be used multiple times

Reply:
and pass just address.name to it, not address